docker-compose.yml

```diff
@@ -0,0 +1,35 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+version: "3.4"
+
+services:
+  standalone:
+    container_name: standalone
+    hostname: localhost
+    image: streamnative/sn-pulsar:2.9.1.1
+    command: >
+      bash -c "bin/apply-config-from-env.py conf/standalone.conf &&
+      exec bin/pulsar standalone -nss -nfw" # disable stream storage and functions worker
+    environment:
+      allowAutoTopicCreationType: partitioned
+      brokerDeleteInactiveTopicsEnabled: "false"
```

Comment:
```suggestion
      PULSAR_PREFIX_allowAutoTopicCreationType: partitioned
      PULSAR_PREFIX_brokerDeleteInactiveTopicsEnabled: "false"
```

Reply:
Here we don't use `PULSAR_PREFIX_` prefix because the property already exists in `conf/standalone.conf`. You can go to the container and see the `conf/standalone.conf` that following properties have been added:

```properties
brokerEntryMetadataInterceptors=org.apache.pulsar.common.intercept.AppendIndexMetadataInterceptor
kafkaAdvertisedListeners=PLAINTEXT://127.0.0.1:19092
kafkaListeners=PLAINTEXT://0.0.0.0:9092
messagingProtocols=kafka
```

While the environment variables without the prefix have been modified in place:

```properties
brokerDeleteInactiveTopicsEnabled=false
allowAutoTopicCreationType=partitioned
```

Reply:
I've pushed a new commit for more explanation about this.

Reply:
IMHO, The `PULSAR_PREFIX_` prefix is used to isolate the environment variables and help us filter the pulsar's environment variables.

```shell
root@localhost:/pulsar/bin# env | grep PULSAR_PREFIX_
PULSAR_PREFIX_messagingProtocols=kafka
PULSAR_PREFIX_brokerDeleteInactiveTopicsEnabled=false
PULSAR_PREFIX_allowAutoTopicCreationType=partitioned
PULSAR_PREFIX_kafkaListeners=PLAINTEXT://0.0.0.0:9092
PULSAR_PREFIX_brokerEntryMetadataInterceptors=org.apache.pulsar.common.intercept.AppendIndexMetadataInterceptor
PULSAR_PREFIX_kafkaAdvertisedListeners=PLAINTEXT://127.0.0.1:19092
```

Reply:
Pulsar's environment variables usually begin with `PULSAR_` prefix, like `PULSAR_EXTRA_CLASSPATH`, `PULSAR_LOG_LEVEL`, `PULSAR_EXTRA_CLASSPATH`.

These environment variables are only used for modifying the config file. See https://github.com/apache/pulsar/blob/f49ed2d9e5afad0ca25505b469aebd01e075f7bd/docker/pulsar/scripts/apply-config-from-env.py#L70-L75